# Data Proffesionals Survey Dataset wrangling & cleaning
### by Kola Ademola

In [1]:
#importing the neccessary libraries

import pandas as pd
import numpy as np

In [2]:
#loading the dataset

data_pro_df = pd.read_excel('Power BI - Final Project.xlsx')

In [3]:
#view dataset

data_pro_df

Unique ID      Email Date Taken (America/New_York)  \
0    62a33b3db4da29969c62df3d  anonymous                     6/10/2022   
1    62a33ba1bae91e4b8b82e35c  anonymous                     6/10/2022   
2    62a33c2cbc6861bf3176bec1  anonymous                     6/10/2022   
3    62a33c8624a26260273822f9  anonymous                     6/10/2022   
4    62a33c91f3072dd892621e03  anonymous                     6/10/2022   
..                        ...        ...                           ...   
625  62b525563f28f20328aeee5c  anonymous                     6/23/2022   
626  62b5a3e29bc428d5345f6e89  anonymous                     6/24/2022   
627  62b71083f31287f32e189026  anonymous                     6/25/2022   
628  62b795033b026e423f287ecd  anonymous                     6/25/2022   
629  62b89039377223ff07b80fb5  anonymous                     6/26/2022   

    Time Taken (America/New_York)  Browser  OS  City  Country  Referrer  \
0                            8:38      NaN NaN   NaN      NaN       NaN   
1                            8:40      NaN NaN   NaN      NaN       NaN   
2                            8:42      NaN NaN   NaN      NaN       NaN   
3                            8:43      NaN NaN   NaN      NaN       NaN   
4                            8:44      NaN NaN   NaN      NaN       NaN   
..                            ...      ...  ..   ...      ...       ...   
625                         22:45      NaN NaN   NaN      NaN       NaN   
626                          7:45      NaN NaN   NaN      NaN       NaN   
627                          9:41      NaN NaN   NaN      NaN       NaN   
628                         19:06      NaN NaN   NaN      NaN       NaN   
629                         12:58      NaN NaN   NaN      NaN       NaN   

    Time Spent  ...  \
0      0:00:44  ...   
1      0:01:30  ...   
2      0:02:18  ...   
3      0:02:10  ...   
4      0:01:51  ...   
..         ...  ...   
625    0:00:50  ...   
626    0:03:12  ...   
627    0:04:43  ...   
628    0:02:17  ...   
629    0:01:19  ...   

    Q6 - How Happy are you in your Current Position with the following? (Management)  \
0                                                  5.0                                 
1                                                  2.0                                 
2                                                  5.0                                 
3                                                 10.0                                 
4                                                  4.0                                 
..                                                 ...                                 
625                                                6.0                                 
626                                                7.0                                 
627                                                2.0                                 
628                                                6.0                                 
629                                                7.0                                 

    Q6 - How Happy are you in your Current Position with the following? (Upward Mobility)  \
0                                                  5.0                                      
1                                                  1.0                                      
2                                                  7.0                                      
3                                                  7.0                                      
4                                                  0.0                                      
..                                                 ...                                      
625                                                5.0                                      
626                                                4.0                                      
627                                                1.0     

In [4]:
#check the basic info on the dataset

data_pro_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 630 entries, 0 to 629
Data columns (total 28 columns):
 #   Column                                                                                        Non-Null Count  Dtype  
---  ------                                                                                        --------------  -----  
 0   Unique ID                                                                                     630 non-null    object 
 1   Email                                                                                         630 non-null    object 
 2   Date Taken (America/New_York)                                                                 630 non-null    object 
 3   Time Taken (America/New_York)                                                                 630 non-null    object 
 4   Browser                                                                                       0 non-null      float64
 5   OS                           

>Datset has **630 rows** & **28 columns**

### Duplicating the dataset before cleaning

In [5]:
#making copy of dataset

df = data_pro_df.copy()

#### Some columns are not neccessary for my analyst, so I will be deleting these columns:
    
   * Email
   * Date Taken (America/New_York)
   * Time Taken (America/New_York)
   * Browser
   * OS
   * City
   * Country
   * Referrer
   * Time Spent
   * Q13 - Ethnicity
   * Q6 - How Happy are you in your Current Position with the following? (Coworkers)
   * Q6 - How Happy are you in your Current Position with the following? (Upward Mobility)	
   * Q6 - How Happy are you in your Current Position with the following? (Learning New Things)

In [6]:
#Dropping columns I wont be needing for my analysis

#list of columns to be dropped
cols = ['Email', 'Date Taken (America/New_York)', 'Time Taken (America/New_York)',
        'Browser', 'OS', 'City', 'Country', 'Referrer', 'Time Spent', 'Q13 - Ethnicity',
        'Q6 - How Happy are you in your Current Position with the following? (Coworkers)',
        'Q6 - How Happy are you in your Current Position with the following? (Upward Mobility)',
        'Q6 - How Happy are you in your Current Position with the following? (Learning New Things)']

for col in cols:
    df = df.drop(col, axis = 1)

In [7]:
#confirm change

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 630 entries, 0 to 629
Data columns (total 15 columns):
 #   Column                                                                                        Non-Null Count  Dtype  
---  ------                                                                                        --------------  -----  
 0   Unique ID                                                                                     630 non-null    object 
 1   Q1 - Which Title Best Fits your Current Role?                                                 630 non-null    object 
 2   Q2 - Did you switch careers into Data?                                                        630 non-null    object 
 3   Q3 - Current Yearly Salary (in USD)                                                           630 non-null    object 
 4   Q4 - What Industry do you work in?                                                            630 non-null    object 
 5   Q5 - Favorite Programming Lan

In [8]:
#view data

df.head()

Unique ID Q1 - Which Title Best Fits your Current Role?  \
0  62a33b3db4da29969c62df3d                                  Data Analyst   
1  62a33ba1bae91e4b8b82e35c                                  Data Analyst   
2  62a33c2cbc6861bf3176bec1                                 Data Engineer   
3  62a33c8624a26260273822f9  Other (Please Specify):Analytics Consultant    
4  62a33c91f3072dd892621e03                                  Data Analyst   

  Q2 - Did you switch careers into Data? Q3 - Current Yearly Salary (in USD)  \
0                                    Yes                           106k-125k   
1                                     No                             41k-65k   
2                                     No                               0-40k   
3                                    Yes                           150k-225k   
4                                    Yes                             41k-65k   

    Q4 - What Industry do you work in? Q5 - Favorite Programming Language  \
0                           Healthcare                             Python   
1                              Finance                                  R   
2  Other (Please Specify):Clean Energy                             Python   
3                              Finance                                  R   
4                           Healthcare                                  R   

   Q6 - How Happy are you in your Current Position with the following? (Salary)  \
0                                                9.0                              
1                                                1.0                              
2                                                0.0                              
3                                               10.0                              
4                                                1.0                              

   Q6 - How Happy are you in your Current Position with the following? (Work/Life Balance)  \
0                                                9.0                                         
1                                                2.0                                         
2                                                8.0                                         
3                                                6.0                                         
4                                                4.0                                         

   Q6 - How Happy are you in your Current Position with the following? (Management)  \
0                                                5.0                                  
1                                                2.0                                  
2                                                5.0                                  
3                                               10.0                                  
4                                                4.0                                  

  Q7 - How difficult was it for you to break into Data?  \
0                                     Very Difficult      
1                                     Very Difficult      
2                                               Easy      
3                                          Very Easy      
4                                          Difficult      

  Q8 - If you were to look for a new job today, what would be the most important thing to you?  \
0                                        Remote Work                                             
1                                        Remote Work                                             
2                             Good Work/Life Balance                                             
3                                        Remote Work                                             
4                                      Better Salary                                             

  Q9 - Male/Female?  Q10 - Current Age Q11 - Which Country do you live in?  \
0 

In [9]:
#check basic statistics of dataset

df.describe().transpose()

count       mean  \
Q6 - How Happy are you in your Current Position...  623.0   4.269663   
Q6 - How Happy are you in your Current Position...  620.0   5.741935   
Q6 - How Happy are you in your Current Position...  618.0   5.326861   
Q10 - Current Age                                   630.0  29.866667   

                                                         std   min   25%  \
Q6 - How Happy are you in your Current Position...  2.772945   0.0   2.0   
Q6 - How Happy are you in your Current Position...  2.761584   0.0   4.0   
Q6 - How Happy are you in your Current Position...  2.738161   0.0   4.0   
Q10 - Current Age                                   7.245941  18.0  25.0   

                                                     50%   75%   max  
Q6 - How Happy are you in your Current Position...   4.0   7.0  10.0  
Q6 - How Happy are you in your Current Position...   6.0   8.0  10.0  
Q6 - How Happy are you in your Current Position...   5.0   7.0  10.0  
Q10 - Current Age                                   28.0  33.0  92.0

### For the 'Current role', I want to standardize their answers and change 'Other Please specify' to just 'Others'

In [10]:
#spliting the Current Role column suing '('

other_role = df['Q1 - Which Title Best Fits your Current Role?'].str.split('(', expand = True)

In [11]:
#replacing the answers with the first part of the split

df['Q1 - Which Title Best Fits your Current Role?'] = other_role[0].str.strip()


In [12]:
#confirm change

df.head()

Unique ID Q1 - Which Title Best Fits your Current Role?  \
0  62a33b3db4da29969c62df3d                                  Data Analyst   
1  62a33ba1bae91e4b8b82e35c                                  Data Analyst   
2  62a33c2cbc6861bf3176bec1                                 Data Engineer   
3  62a33c8624a26260273822f9                                         Other   
4  62a33c91f3072dd892621e03                                  Data Analyst   

  Q2 - Did you switch careers into Data? Q3 - Current Yearly Salary (in USD)  \
0                                    Yes                           106k-125k   
1                                     No                             41k-65k   
2                                     No                               0-40k   
3                                    Yes                           150k-225k   
4                                    Yes                             41k-65k   

    Q4 - What Industry do you work in? Q5 - Favorite Programming Language  \
0                           Healthcare                             Python   
1                              Finance                                  R   
2  Other (Please Specify):Clean Energy                             Python   
3                              Finance                                  R   
4                           Healthcare                                  R   

   Q6 - How Happy are you in your Current Position with the following? (Salary)  \
0                                                9.0                              
1                                                1.0                              
2                                                0.0                              
3                                               10.0                              
4                                                1.0                              

   Q6 - How Happy are you in your Current Position with the following? (Work/Life Balance)  \
0                                                9.0                                         
1                                                2.0                                         
2                                                8.0                                         
3                                                6.0                                         
4                                                4.0                                         

   Q6 - How Happy are you in your Current Position with the following? (Management)  \
0                                                5.0                                  
1                                                2.0                                  
2                                                5.0                                  
3                                               10.0                                  
4                                                4.0                                  

  Q7 - How difficult was it for you to break into Data?  \
0                                     Very Difficult      
1                                     Very Difficult      
2                                               Easy      
3                                          Very Easy      
4                                          Difficult      

  Q8 - If you were to look for a new job today, what would be the most important thing to you?  \
0                                        Remote Work                                             
1                                        Remote Work                                             
2                             Good Work/Life Balance                                             
3                                        Remote Work                                             
4                                      Better Salary                                             

  Q9 - Male/Female?  Q10 - Current Age Q11 - Which Country do you live in?  \
0 

In [13]:
#changing 'Other' to 'Others'

df['Q1 - Which Title Best Fits your Current Role?'] = df['Q1 - Which Title Best Fits your Current Role?'].str.replace('Other', 'Others')

#confirm change

df.head()

Unique ID Q1 - Which Title Best Fits your Current Role?  \
0  62a33b3db4da29969c62df3d                                  Data Analyst   
1  62a33ba1bae91e4b8b82e35c                                  Data Analyst   
2  62a33c2cbc6861bf3176bec1                                 Data Engineer   
3  62a33c8624a26260273822f9                                        Others   
4  62a33c91f3072dd892621e03                                  Data Analyst   

  Q2 - Did you switch careers into Data? Q3 - Current Yearly Salary (in USD)  \
0                                    Yes                           106k-125k   
1                                     No                             41k-65k   
2                                     No                               0-40k   
3                                    Yes                           150k-225k   
4                                    Yes                             41k-65k   

    Q4 - What Industry do you work in? Q5 - Favorite Programming Language  \
0                           Healthcare                             Python   
1                              Finance                                  R   
2  Other (Please Specify):Clean Energy                             Python   
3                              Finance                                  R   
4                           Healthcare                                  R   

   Q6 - How Happy are you in your Current Position with the following? (Salary)  \
0                                                9.0                              
1                                                1.0                              
2                                                0.0                              
3                                               10.0                              
4                                                1.0                              

   Q6 - How Happy are you in your Current Position with the following? (Work/Life Balance)  \
0                                                9.0                                         
1                                                2.0                                         
2                                                8.0                                         
3                                                6.0                                         
4                                                4.0                                         

   Q6 - How Happy are you in your Current Position with the following? (Management)  \
0                                                5.0                                  
1                                                2.0                                  
2                                                5.0                                  
3                                               10.0                                  
4                                                4.0                                  

  Q7 - How difficult was it for you to break into Data?  \
0                                     Very Difficult      
1                                     Very Difficult      
2                                               Easy      
3                                          Very Easy      
4                                          Difficult      

  Q8 - If you were to look for a new job today, what would be the most important thing to you?  \
0                                        Remote Work                                             
1                                        Remote Work                                             
2                             Good Work/Life Balance                                             
3                                        Remote Work                                             
4                                      Better Salary                                             

  Q9 - Male/Female?  Q10 - Current Age Q11 - Which Country do you live in?  \
0 

### For the Salary I want to split the column, convert it to integer and get average salary from that

In [14]:
#splitting the salary column by '-' and remove the 'k' & '+'

#split salary column on '-' and remove the 'k'
salary = df['Q3 - Current Yearly Salary (in USD)'].str.replace('k', '').str.split('-', expand = True).fillna(np.NaN)

#removing the '+'
salary[0] = salary[0].str.replace('+', '')

#fill the null rows with '225' for rows with '225+' initially so I can get the average
salary[1] = salary[1].fillna(salary[0])

C:\Users\Phantom\AppData\Local\Temp\ipykernel_6424\515457618.py:7: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  salary[0] = salary[0].str.replace('+', '')


In [15]:
#convert the salaries from object to int so I can erform mathematical operations

cols = [0, 1]

for col in cols:
    salary[col] = salary[col].astype(int)
    

#get the average of the salaries
salary['avg'] = (salary[0] + salary[1]) / 2

#view the new salary subset
salary.head()

0    1    avg
0  106  125  115.5
1   41   65   53.0
2    0   40   20.0
3  150  225  187.5
4   41   65   53.0

In [16]:
#assign the avg salary to its own column in the main dataset

df['Average Salary'] = salary['avg']

#view dataset
df.head()

Unique ID Q1 - Which Title Best Fits your Current Role?  \
0  62a33b3db4da29969c62df3d                                  Data Analyst   
1  62a33ba1bae91e4b8b82e35c                                  Data Analyst   
2  62a33c2cbc6861bf3176bec1                                 Data Engineer   
3  62a33c8624a26260273822f9                                        Others   
4  62a33c91f3072dd892621e03                                  Data Analyst   

  Q2 - Did you switch careers into Data? Q3 - Current Yearly Salary (in USD)  \
0                                    Yes                           106k-125k   
1                                     No                             41k-65k   
2                                     No                               0-40k   
3                                    Yes                           150k-225k   
4                                    Yes                             41k-65k   

    Q4 - What Industry do you work in? Q5 - Favorite Programming Language  \
0                           Healthcare                             Python   
1                              Finance                                  R   
2  Other (Please Specify):Clean Energy                             Python   
3                              Finance                                  R   
4                           Healthcare                                  R   

   Q6 - How Happy are you in your Current Position with the following? (Salary)  \
0                                                9.0                              
1                                                1.0                              
2                                                0.0                              
3                                               10.0                              
4                                                1.0                              

   Q6 - How Happy are you in your Current Position with the following? (Work/Life Balance)  \
0                                                9.0                                         
1                                                2.0                                         
2                                                8.0                                         
3                                                6.0                                         
4                                                4.0                                         

   Q6 - How Happy are you in your Current Position with the following? (Management)  \
0                                                5.0                                  
1                                                2.0                                  
2                                                5.0                                  
3                                               10.0                                  
4                                                4.0                                  

  Q7 - How difficult was it for you to break into Data?  \
0                                     Very Difficult      
1                                     Very Difficult      
2                                               Easy      
3                                          Very Easy      
4                                          Difficult      

  Q8 - If you were to look for a new job today, what would be the most important thing to you?  \
0                                        Remote Work                                             
1                                        Remote Work                                             
2                             Good Work/Life Balance                                             
3                                        Remote Work                                             
4                                      Better Salary                                             

  Q9 - Male/Female?  Q10 - Current Age Q11 - Which Country do you live in?  \
0 

### I will be extracting all corrected spelled countries from the country column

In [17]:
#standardize all answers as proper case for consistency

df['Q11 - Which Country do you live in?'] = df['Q11 - Which Country do you live in?'].str.title()

#removing the 'Other (Please Specify):'
country = df['Q11 - Which Country do you live in?'].str.split(':', expand = True).fillna(np.NaN)

#view data
country.head()

0        1
0           United States      NaN
1                  Canada      NaN
2  Other (Please Specify)  Nigeria
3                  Canada      NaN
4           United States      NaN

In [18]:
#Filling the null values with the country names from the left side
country[1] = country[1].fillna(country[0])

#confirm change
country.head()

0              1
0           United States  United States
1                  Canada         Canada
2  Other (Please Specify)        Nigeria
3                  Canada         Canada
4           United States  United States

In [19]:
#I used the new country column to fill/replace the countries in the main data
df['Q11 - Which Country do you live in?'] = country[1]

#view dataset
df.head()

Unique ID Q1 - Which Title Best Fits your Current Role?  \
0  62a33b3db4da29969c62df3d                                  Data Analyst   
1  62a33ba1bae91e4b8b82e35c                                  Data Analyst   
2  62a33c2cbc6861bf3176bec1                                 Data Engineer   
3  62a33c8624a26260273822f9                                        Others   
4  62a33c91f3072dd892621e03                                  Data Analyst   

  Q2 - Did you switch careers into Data? Q3 - Current Yearly Salary (in USD)  \
0                                    Yes                           106k-125k   
1                                     No                             41k-65k   
2                                     No                               0-40k   
3                                    Yes                           150k-225k   
4                                    Yes                             41k-65k   

    Q4 - What Industry do you work in? Q5 - Favorite Programming Language  \
0                           Healthcare                             Python   
1                              Finance                                  R   
2  Other (Please Specify):Clean Energy                             Python   
3                              Finance                                  R   
4                           Healthcare                                  R   

   Q6 - How Happy are you in your Current Position with the following? (Salary)  \
0                                                9.0                              
1                                                1.0                              
2                                                0.0                              
3                                               10.0                              
4                                                1.0                              

   Q6 - How Happy are you in your Current Position with the following? (Work/Life Balance)  \
0                                                9.0                                         
1                                                2.0                                         
2                                                8.0                                         
3                                                6.0                                         
4                                                4.0                                         

   Q6 - How Happy are you in your Current Position with the following? (Management)  \
0                                                5.0                                  
1                                                2.0                                  
2                                                5.0                                  
3                                               10.0                                  
4                                                4.0                                  

  Q7 - How difficult was it for you to break into Data?  \
0                                     Very Difficult      
1                                     Very Difficult      
2                                               Easy      
3                                          Very Easy      
4                                          Difficult      

  Q8 - If you were to look for a new job today, what would be the most important thing to you?  \
0                                        Remote Work                                             
1                                        Remote Work                                             
2                             Good Work/Life Balance                                             
3                                        Remote Work                                             
4                                      Better Salary                                             

  Q9 - Male/Female?  Q10 - Current Age Q11 - Which Country do you live in?  \
0 

### I will be changing the 'Others (Please Specify)' to just 'Others' for the Industry the participants worked in since its not relevant to me for this analysis

In [20]:
#splitting column on '('
industry = df['Q4 - What Industry do you work in?'].str.split('(', expand = True)

#view subset data
industry

0                              1     2
0    Healthcare                           None  None
1       Finance                           None  None
2        Other    Please Specify):Clean Energy  None
3       Finance                           None  None
4    Healthcare                           None  None
..          ...                            ...   ...
625      Other          Please Specify):Retail  None
626      Other   Please Specify):Manufacturing  None
627        Tech                           None  None
628        Tech                           None  None
629      Other      Please Specify):Automotive  None

[630 rows x 3 columns]

In [21]:
#replace the new answers in the main dataset and change 'Other' to 'Others'
df['Q4 - What Industry do you work in?'] = industry[0].str.strip().replace('Other', 'Others')

df.head()

Unique ID Q1 - Which Title Best Fits your Current Role?  \
0  62a33b3db4da29969c62df3d                                  Data Analyst   
1  62a33ba1bae91e4b8b82e35c                                  Data Analyst   
2  62a33c2cbc6861bf3176bec1                                 Data Engineer   
3  62a33c8624a26260273822f9                                        Others   
4  62a33c91f3072dd892621e03                                  Data Analyst   

  Q2 - Did you switch careers into Data? Q3 - Current Yearly Salary (in USD)  \
0                                    Yes                           106k-125k   
1                                     No                             41k-65k   
2                                     No                               0-40k   
3                                    Yes                           150k-225k   
4                                    Yes                             41k-65k   

  Q4 - What Industry do you work in? Q5 - Favorite Programming Language  \
0                         Healthcare                             Python   
1                            Finance                                  R   
2                             Others                             Python   
3                            Finance                                  R   
4                         Healthcare                                  R   

   Q6 - How Happy are you in your Current Position with the following? (Salary)  \
0                                                9.0                              
1                                                1.0                              
2                                                0.0                              
3                                               10.0                              
4                                                1.0                              

   Q6 - How Happy are you in your Current Position with the following? (Work/Life Balance)  \
0                                                9.0                                         
1                                                2.0                                         
2                                                8.0                                         
3                                                6.0                                         
4                                                4.0                                         

   Q6 - How Happy are you in your Current Position with the following? (Management)  \
0                                                5.0                                  
1                                                2.0                                  
2                                                5.0                                  
3                                               10.0                                  
4                                                4.0                                  

  Q7 - How difficult was it for you to break into Data?  \
0                                     Very Difficult      
1                                     Very Difficult      
2                                               Easy      
3                                          Very Easy      
4                                          Difficult      

  Q8 - If you were to look for a new job today, what would be the most important thing to you?  \
0                                        Remote Work                                             
1                                        Remote Work                                             
2                             Good Work/Life Balance                                             
3                                        Remote Work                                             
4                                      Better Salary                                             

  Q9 - Male/Female?  Q10 - Current Age Q11 - Which Country do you live in?  \
0             

In [22]:
df['Q4 - What Industry do you work in?'].value_counts()

Others               205
Tech                 150
Finance               97
Healthcare            84
Education             38
Telecommunication     22
Construction          14
Real Estate           13
Agriculture            7
Name: Q4 - What Industry do you work in?, dtype: int64

### I will changing the open ended answers for Q8 to just 'Others'

In [23]:
#splitting the column on '('
q8 = df['Q8 - If you were to look for a new job today, what would be the most important thing to you?'].str.split('(', expand = True)

#view subset
q8

0                                    1
0               Remote Work                                 None
1               Remote Work                                 None
2    Good Work/Life Balance                                 None
3               Remote Work                                 None
4             Better Salary                                 None
..                      ...                                  ...
625           Better Salary                                 None
626           Better Salary                                 None
627           Better Salary                                 None
628                  Other   Please Specify):Career Advancedment
629             Remote Work                                 None

[630 rows x 2 columns]

In [24]:
#replacing the new answers in the main data and changing 'Other' to 'Others'

df['Q8 - If you were to look for a new job today, what would be the most important thing to you?'] = q8[0].str.strip().replace('Other', 'Others')

#confirm change
df.head()

Unique ID Q1 - Which Title Best Fits your Current Role?  \
0  62a33b3db4da29969c62df3d                                  Data Analyst   
1  62a33ba1bae91e4b8b82e35c                                  Data Analyst   
2  62a33c2cbc6861bf3176bec1                                 Data Engineer   
3  62a33c8624a26260273822f9                                        Others   
4  62a33c91f3072dd892621e03                                  Data Analyst   

  Q2 - Did you switch careers into Data? Q3 - Current Yearly Salary (in USD)  \
0                                    Yes                           106k-125k   
1                                     No                             41k-65k   
2                                     No                               0-40k   
3                                    Yes                           150k-225k   
4                                    Yes                             41k-65k   

  Q4 - What Industry do you work in? Q5 - Favorite Programming Language  \
0                         Healthcare                             Python   
1                            Finance                                  R   
2                             Others                             Python   
3                            Finance                                  R   
4                         Healthcare                                  R   

   Q6 - How Happy are you in your Current Position with the following? (Salary)  \
0                                                9.0                              
1                                                1.0                              
2                                                0.0                              
3                                               10.0                              
4                                                1.0                              

   Q6 - How Happy are you in your Current Position with the following? (Work/Life Balance)  \
0                                                9.0                                         
1                                                2.0                                         
2                                                8.0                                         
3                                                6.0                                         
4                                                4.0                                         

   Q6 - How Happy are you in your Current Position with the following? (Management)  \
0                                                5.0                                  
1                                                2.0                                  
2                                                5.0                                  
3                                               10.0                                  
4                                                4.0                                  

  Q7 - How difficult was it for you to break into Data?  \
0                                     Very Difficult      
1                                     Very Difficult      
2                                               Easy      
3                                          Very Easy      
4                                          Difficult      

  Q8 - If you were to look for a new job today, what would be the most important thing to you?  \
0                                        Remote Work                                             
1                                        Remote Work                                             
2                             Good Work/Life Balance                                             
3                                        Remote Work                                             
4                                      Better Salary                                             

  Q9 - Male/Female?  Q10 - Current Age Q11 - Which Country do you live in?  \
0             

### I will be filtering just the countries with the correct spelling, I will do this by importing a dataset with the list of countries worldwide and the I will put the names of the countries in a list, its the list I will use as the criteria for filtering.

In [25]:
#importing the new dataset with the list of countries
countries = pd.read_csv('countries.csv')

#view dataset
countries

Countries
0       Afghanistan
1           Albania
2           Algeria
3           Andorra
4            Angola
..              ...
188         Vietnam
189  Western Sahara
190           Yemen
191          Zambia
192        Zimbabwe

[193 rows x 1 columns]

In [26]:
#putting the names of the countries in a list
country = []
num = countries.shape[0]
for i in range(0, num):
    country.append(countries['Countries'][i])

#view list
country

['Afghanistan',
 'Albania',
 'Algeria',
 'Andorra',
 'Angola',
 'Antigua and Barbuda',
 'Argentina',
 'Armenia',
 'Austria',
 'Azerbaijan',
 'Bahrain',
 'Bangladesh',
 'Barbados',
 'Belarus',
 'Belgium',
 'Belize',
 'Benin',
 'Bhutan',
 'Bolivia',
 'Bosnia and Herzegovina',
 'Botswana',
 'Brazil',
 'Brunei',
 'Bulgaria',
 'Burkina Faso',
 'Burundi',
 'Cabo Verde',
 'Cambodia',
 'Cameroon',
 'Canada',
 'Central African Republic',
 'Chad',
 'Channel Islands',
 'Chile',
 'China',
 'Colombia',
 'Comoros',
 'Congo',
 'Costa Rica',
 "Côte d'Ivoire",
 'Croatia',
 'Cuba',
 'Cyprus',
 'Czech Republic',
 'Denmark',
 'Djibouti',
 'Dominica',
 'Dominican Republic',
 'DR Congo',
 'Ecuador',
 'Egypt',
 'El Salvador',
 'Equatorial Guinea',
 'Eritrea',
 'Estonia',
 'Eswatini',
 'Ethiopia',
 'Faeroe Islands',
 'Finland',
 'France',
 'French Guiana',
 'Gabon',
 'Gambia',
 'Georgia',
 'Germany',
 'Ghana',
 'Gibraltar',
 'Greece',
 'Grenada',
 'Guatemala',
 'Guinea',
 'Guinea-Bissau',
 'Guyana',
 'Haiti',

In [27]:
#filtering the dataset using the names of countries
df = df[df['Q11 - Which Country do you live in?'].isin(country)]

In [28]:
#checking info of dataset for any errors
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 575 entries, 0 to 629
Data columns (total 16 columns):
 #   Column                                                                                        Non-Null Count  Dtype  
---  ------                                                                                        --------------  -----  
 0   Unique ID                                                                                     575 non-null    object 
 1   Q1 - Which Title Best Fits your Current Role?                                                 575 non-null    object 
 2   Q2 - Did you switch careers into Data?                                                        575 non-null    object 
 3   Q3 - Current Yearly Salary (in USD)                                                           575 non-null    object 
 4   Q4 - What Industry do you work in?                                                            575 non-null    object 
 5   Q5 - Favorite Programming Lan

### Done with cleaning and ready to export back as excel file to use for visualization in PowerBI

In [29]:
df.to_excel('survey_data_cleaned.xlsx', sheet_name = 'Data Profeessionals Survey', index = False)